src/main/java/com/palantir/gradle/graal/NativeImageTask.java

```diff
@@ -37,16 +35,37 @@
 
     public NativeImageTask() {
         setDescription("Runs GraalVM's native-image command with configured options and parameters.");
+
         // must use an anonymous inner class instead of a lambda to get Gradle staleness checking
         doLast(new LogAction());
     }
 
+    /**
+     * Returns a platform-dependent file extension for executables.
+     *
+     * @return an empty String on {@link Platform.OperatingSystem#MAC MAC} and
+     *         {@link Platform.OperatingSystem#LINUX LINUX}, ".exe" on {@link Platform.OperatingSystem#WINDOWS WINDOWS}
+     */
+    @Override
+    protected String getArchitectureSpecifiedOutputExtension() {
+        switch (Platform.operatingSystem()) {
+            case MAC:
+            case LINUX:
+                return "";
+            case WINDOWS:
+                return ".exe";
+            default:
+                throw new IllegalStateException("No GraalVM support for " + Platform.operatingSystem());
+        }
+    }
+
     @TaskAction
     public final void nativeImage() throws IOException {
         List<String> args = new ArrayList<>();
         configureArgs(args);
         args.add(mainClass.get());
         getProject().exec(spec -> {
+            configurePlatformSpecifics(spec);
```

Comment:
idle thought: could we make this return a spec, and then change the setup here to do something like:
```java
getProject().exec(withPlatformCustomizedEnvironment(spec -> {
...
});
```

Maybe feels a bit more ergonomic because we're not depending on the side-effects of the method call?

Reply:
Good idea, but same as [above](https://github.com/palantir/gradle-graal/pull/127#discussion_r315865928), might be a bigger change.